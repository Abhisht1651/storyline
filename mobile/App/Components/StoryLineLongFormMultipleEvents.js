// @flow

import React, { useState } from 'react'
import PropTypes from 'prop-types'
import { View, Text, Image, TouchableOpacity, ImageBackground, ImageProps } from 'react-native'
import Carousel from 'react-native-snap-carousel'
import { widthPercentageToDP as wp } from '../Themes/ScreenMetrics'

// Styles
import styles from './Styles/StoryLineLongFormMultipleEventsStyle'
import images from '../Themes/Images'
import Timeline from './Timeline'

// Components
import { X4SButton } from './Buttons/X4SButton'
import { DefaultButton } from './Buttons/DefaultButton'

type Props = {
  reach: string,
  friendsFollowing: string,
  category: string,
  endStatus: string,
  storyTitle: string,
  updatedTime: string,
  followDisplay: boolean,
  following: boolean,
  trending: boolean,
  active: boolean,
  date: string,
  totalStoryItems: number,
  carouselItems: Array<{
    previewText: string,
    commentText: string,
    storyImage: ImageProps,
    read: boolean
  }>,
  onPressShare: PropTypes.func,
  onPressReact: PropTypes.func,
  onPressRaven: PropTypes.func,
  onPressFollow: PropTypes.func
}

const StoryLineLongFormMultipleEvents = (props: Props) => {
  const [paginationActive, setPaginationActive] = useState(0)

  const {
    friendsFollowing,
    reach,
    endStatus,
    storyTitle,
    updatedTime,
    category,
    date,
    totalStoryItems,
    carouselItems,
    followDisplay,
    following,
    trending,
    active,
    onPressReact,
    onPressRaven,
    onPressShare,
    onPressFollow
  } = props

  const {
    iconStoryLineTrending,
    pollActiveIcon,
    iconStoryLineReach,
    iconStoryLineRelatedEntity1,
    iconStoryLineRelatedEntity2,
    iconStoryLineRelatedEntity3,
    iconStoryLineRelatedEntity4,
    voterIcon1,
    voterIcon2,
    voterIcon3,
    shareIcon,
    reactIcon,
    ravenIcon,
    sourceIcon,
    backIcon,
    iconPeople
  } = images

  const iconRelatedEntity = [
    iconStoryLineRelatedEntity1,
    iconStoryLineRelatedEntity2,
    iconStoryLineRelatedEntity3,
    iconStoryLineRelatedEntity4
  ]

  const voterIcons = [voterIcon1, voterIcon2, voterIcon3]

  const commentIconsElement = voterIcons.map((voterIcon, index) => (
    <Image source={voterIcon} key={index} style={styles.commentsIcon} />
  ))

  const relatedEntityIconContainer = iconRelatedEntity.map((icon, index) => (
    <Image source={icon} key={index} style={styles.relatedEntityIcon} />
  ))

  function renderItem({
    item,
    index
  }: {
    item: { previewText: string, commentText: string, storyImage: ImageProps, read: boolean },
    index: number
  }) {
    const { previewText, commentText, storyImage, read } = item
    if (read) {
      return (
        <View key={index} style={styles.previewContainerRead}>
          <Text style={styles.readText}>Read full storyline</Text>
          <Image source={backIcon} style={styles.backIcon} />
        </View>
      )
    }
    return (
      <View key={index} style={styles.previewContainer}>
        <ImageBackground source={storyImage} style={styles.image} imageStyle={styles.imageBorder}>
          <View style={styles.imageItems}>
            {index === paginationActive ? (
              <View style={styles.playButton}>
                <View style={styles.triangle} />
              </View>
            ) : null}
            {index === paginationActive ? (
              <Image source={sourceIcon} style={styles.sourceImage} />
            ) : null}
          </View>
        </ImageBackground>
        <Text style={styles.previewText}>{previewText}</Text>
        <View style={styles.relatedEntityContainer}>
          <Text style={styles.EntityRelatedText}>Related entities</Text>
          <View style={styles.relatedEntityIconContainer}>{relatedEntityIconContainer}</View>
        </View>
        <View style={styles.divider} />
        <View style={styles.commentsContainer}>
          {commentIconsElement}
          <Text style={styles.commentsText}>{commentText}</Text>
        </View>
      </View>
    )
  }

  return (
    <View style={styles.container}>
      <View style={styles.headerContainer}>
        <View style={styles.headerLeft}>
          <Text style={styles.headingText}>{category}</Text>
          <View style={styles.dot} />
          <Image source={pollActiveIcon} style={styles.storyLineIconHeading} />
          {active ? <Text style={styles.activeText}>Active</Text> : null}
          {trending ? <View style={styles.dot} /> : null}
          {trending ? (
            <Image source={iconStoryLineTrending} style={styles.storyLineIconHeading} />
          ) : null}
          <View style={styles.endedButton}>
            <Text style={styles.endedButtonText}>{endStatus}</Text>
          </View>
        </View>
        {followDisplay ? (
          <TouchableOpacity>
            {following ? (
              <X4SButton disabled title="Following" onPress={onPressFollow} />
            ) : (
              <DefaultButton outline title="Follow" onPress={onPressFollow} />
            )}
          </TouchableOpacity>
        ) : null}
      </View>
      <View style={styles.friends}>
        <View style={styles.friendsTextContainer}>
          <Image source={iconStoryLineReach} style={styles.friendsIcon} />
          <Text style={styles.friendsText}> {reach}</Text>
        </View>
        <View style={styles.friendsTextContainer}>
          <Image source={iconPeople} style={styles.friendsIcon} />
          <Text style={styles.friendsText}> {friendsFollowing}</Text>
        </View>
      </View>
      <Text style={styles.newsTitleText}>{storyTitle}</Text>
      <Text style={styles.updateTimeText}>Updated {updatedTime} ago</Text>
      <View style={styles.caraouselContainer}>
        <Carousel
          containerCustomStyle={styles.slider}
          contentContainerCustomStyle={styles.sliderContentContainer}
          data={carouselItems}
          inactiveSlideOpacity={1}
          inactiveSlideScale={1}
          sliderWidth={wp('86.67%%')}
          itemWidth={wp('77.86%')}
          renderItem={renderItem}
          onSnapToItem={(index) => setPaginationActive(index)}
        />
      </View>
      <Text style={styles.date}>{date}</Text>
      <Timeline len={totalStoryItems} active={paginationActive} />
      <View style={styles.actionIconsContainer}>
        <TouchableOpacity onPress={onPressShare}>
          <Image source={shareIcon} style={styles.actionIcon} />
        </TouchableOpacity>
        <TouchableOpacity onPress={onPressRaven}>
          <Image source={ravenIcon} style={styles.actionIcon} />
        </TouchableOpacity>
        <TouchableOpacity onPress={onPressReact}>
          <Image source={reactIcon} style={styles.actionIcon} />
        </TouchableOpacity>
      </View>
    </View>
  )
}

StoryLineLongFormMultipleEvents.defaultProps = {
  reach: '45K Reached',
  friendsFollowing: '241K Followers',
  category: 'World',
  endStatus: 'Ended',
  storyTitle: 'UK exit from the European Union',
  updatedTime: '2m',
  followDisplay: true,
  following: false,
  trending: true,
  active: false,
  date: 'July 25th,1991',
  totalStoryItems: 8,
  carouselItems: [
    {
      storyImage: images.storyImage,
      previewText: 'Corbyn wins Labor conference Brexit vote in the latest polls',
      commentText: '34K comments',
      read: false
    },
    {
      storyImage: images.storyImage,
      previewText: 'Corbyn wins Labor conference Brexit vote in the latest polls',
      commentText: '34K comments',
      read: false
    },
    {
      storyImage: images.storyImage,
      previewText: 'Corbyn wins Labor conference Brexit vote in the latest polls',
      commentText: '34K comments',
      read: false
    },
    {
      storyImage: null,
      previewText: '',
      commentText: '',
      read: true
    }
  ],
  onPressReact: () => {},
  onPressRaven: () => {},
  onPressShare: () => {},
  onPressFollow: () => {}
}

export default StoryLineLongFormMultipleEvents
