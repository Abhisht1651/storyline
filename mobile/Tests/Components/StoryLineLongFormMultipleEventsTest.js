// @flow
import 'react-native'
import React from 'react'
import { shallow } from 'enzyme'
import StoryLineLongFormMultipleEvents from '../../App/Components/StoryLineLongFormMultipleEvents'

describe('StoryLineLongFormMultipleeEvents', () => {
  it('should render the component itself correctly', () => {
    const wrapper = shallow(<StoryLineLongFormMultipleEvents />)
    expect(wrapper).toMatchSnapshot()
  })
  it('should render the Text Component', () => {
    const wrapper = shallow(<StoryLineLongFormMultipleEvents />)
    expect(wrapper.find('Text')).toHaveLength(7)
  })
  it('should render the Image Component', () => {
    const wrapper = shallow(<StoryLineLongFormMultipleEvents />)
    expect(wrapper.find('Image')).toHaveLength(7)
  })
})
