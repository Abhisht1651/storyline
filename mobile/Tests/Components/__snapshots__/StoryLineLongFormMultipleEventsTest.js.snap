// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StoryLineLongFormMultipleeEvents should render the component itself correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "borderColor": "#E6ECED",
      "borderRadius": 40,
      "borderStyle": "solid",
      "borderWidth": 2,
      "height": 1240,
      "opacity": 1,
      "width": 650,
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginTop": 27,
        "opacity": 1,
        "width": "87.69%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#1F1F60",
            "fontFamily": "BananaGrotesk-Regular",
            "fontSize": 28,
            "letterSpacing": -0.08,
            "opacity": 1,
            "textAlign": "left",
          }
        }
      >
        World
      </Text>
      <View
        style={
          Object {
            "backgroundColor": "#1F1F60",
            "borderRadius": 2,
            "height": 4,
            "marginHorizontal": 10,
            "width": 4,
          }
        }
      />
      <Image
        source={
          Object {
            "default": "",
          }
        }
        style={
          Object {
            "height": 34,
            "width": 34,
          }
        }
      />
      <View
        style={
          Object {
            "backgroundColor": "#1F1F60",
            "borderRadius": 2,
            "height": 4,
            "marginHorizontal": 10,
            "width": 4,
          }
        }
      />
      <Image
        source={
          Object {
            "default": "",
          }
        }
        style={
          Object {
            "height": 34,
            "width": 34,
          }
        }
      />
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#8F8FAF",
            "borderColor": "#E6ECED",
            "borderRadius": 8,
            "borderStyle": "solid",
            "borderWidth": 2,
            "height": 40,
            "justifyContent": "center",
            "marginLeft": 14,
            "opacity": 1,
            "width": 108,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#FFFFFF",
              "fontFamily": "BananaGrotesk-Regular",
              "fontSize": 24,
              "letterSpacing": -0.07,
              "opacity": 1,
              "textAlign": "center",
              "textAlignVertical": "center",
            }
          }
        >
          Ended
        </Text>
      </View>
    </View>
    <ForwardRef>
      <DefaultButton
        disabled={false}
        onPress={[Function]}
        outline={true}
        style={Object {}}
        title="Follow"
      />
    </ForwardRef>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "marginTop": 5,
        "width": "87.69%",
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <Image
        source={
          Object {
            "default": "",
          }
        }
        style={
          Object {
            "height": 26,
            "width": 26,
          }
        }
      />
      <Text
        style={
          Object {
            "color": "#8F8FAF",
            "fontFamily": "BananaGrotesk-Regular",
            "fontSize": 24,
            "letterSpacing": -0.07,
            "opacity": 1,
            "paddingRight": 20,
            "textAlign": "left",
          }
        }
      >
         
        45K Reached
      </Text>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <Image
        source={
          Object {
            "default": "",
          }
        }
        style={
          Object {
            "height": 26,
            "width": 26,
          }
        }
      />
      <Text
        style={
          Object {
            "color": "#8F8FAF",
            "fontFamily": "BananaGrotesk-Regular",
            "fontSize": 24,
            "letterSpacing": -0.07,
            "opacity": 1,
            "paddingRight": 20,
            "textAlign": "left",
          }
        }
      >
         
        241K Followers
      </Text>
    </View>
  </View>
  <Text
    style={
      Object {
        "color": "#1F1F60",
        "fontFamily": "BananaGrotesk-Medium",
        "fontSize": 32,
        "letterSpacing": -0.11,
        "marginTop": 24,
        "opacity": 1,
        "textAlign": "left",
        "width": "87.69%",
      }
    }
  >
    UK exit from the European Union
  </Text>
  <Text
    style={
      Object {
        "color": "#8F8FAF",
        "fontFamily": "BananaGrotesk-Regular",
        "fontSize": 24,
        "letterSpacing": -0.07,
        "marginTop": 5,
        "opacity": 1,
        "textAlign": "left",
        "width": "87.69%",
      }
    }
  >
    Updated 
    2m
     ago
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "marginBottom": 28.5,
        "marginTop": 26,
        "width": "100%",
      }
    }
  >
    <Carousel
      activeAnimationOptions={null}
      activeAnimationType="timing"
      activeSlideAlignment="center"
      activeSlideOffset={20}
      apparitionDelay={0}
      autoplay={false}
      autoplayDelay={1000}
      autoplayInterval={3000}
      callbackOffsetMargin={5}
      containerCustomStyle={Object {}}
      contentContainerCustomStyle={Object {}}
      data={
        Array [
          Object {
            "commentText": "34K comments",
            "previewText": "Corbyn wins Labor conference Brexit vote in the latest polls",
            "read": false,
            "storyImage": Object {
              "default": "",
            },
          },
          Object {
            "commentText": "34K comments",
            "previewText": "Corbyn wins Labor conference Brexit vote in the latest polls",
            "read": false,
            "storyImage": Object {
              "default": "",
            },
          },
          Object {
            "commentText": "34K comments",
            "previewText": "Corbyn wins Labor conference Brexit vote in the latest polls",
            "read": false,
            "storyImage": Object {
              "default": "",
            },
          },
          Object {
            "commentText": "",
            "previewText": "",
            "read": true,
            "storyImage": null,
          },
        ]
      }
      enableMomentum={false}
      enableSnap={true}
      firstItem={0}
      hasParallaxImages={false}
      inactiveSlideOpacity={1}
      inactiveSlideScale={1}
      inactiveSlideShift={0}
      itemWidth={584}
      layout="default"
      lockScrollTimeoutDuration={1000}
      lockScrollWhileSnapping={false}
      loop={false}
      loopClonesPerSide={3}
      onSnapToItem={[Function]}
      renderItem={[Function]}
      scrollEnabled={true}
      shouldOptimizeUpdates={true}
      slideStyle={Object {}}
      sliderWidth={650}
      swipeThreshold={20}
      useScrollView={false}
      vertical={false}
    />
  </View>
  <Text
    style={
      Object {
        "color": "#1F1F60",
        "fontFamily": "BananaGrotesk-Regular",
        "fontSize": 24.5,
        "letterSpacing": 0,
        "marginBottom": 15,
        "opacity": 1,
        "textAlign": "center",
      }
    }
  >
    July 25th,1991
  </Text>
  <timeline
    active={0}
    len={8}
  />
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-around",
        "marginBottom": "7%",
        "marginTop": "8%",
        "width": "100%",
      }
    }
  >
    <ForwardRef
      onPress={[Function]}
    >
      <Image
        source={
          Object {
            "default": "",
          }
        }
        style={
          Object {
            "height": 36,
            "opacity": 1,
            "width": 130,
          }
        }
      />
    </ForwardRef>
    <ForwardRef
      onPress={[Function]}
    >
      <Image
        source={
          Object {
            "default": "",
          }
        }
        style={
          Object {
            "height": 36,
            "opacity": 1,
            "width": 130,
          }
        }
      />
    </ForwardRef>
    <ForwardRef
      onPress={[Function]}
    >
      <Image
        source={
          Object {
            "default": "",
          }
        }
        style={
          Object {
            "height": 36,
            "opacity": 1,
            "width": 130,
          }
        }
      />
    </ForwardRef>
  </View>
</View>
`;
